### ISSUE TITLE
MUV data discrepancy

### ISSUE THREAD

**Issue by egracheva** [deepchem/deepchem#3914]
## 🐛 Data discrepancy

When downloading the MUV dataset from the official MoleculeNet page, I get a dataset with many NaN target values (y).
When loading the same data with the `load_muv` function, all the NaNs are shown as zeroes.
Obviously, the data is not the same; I wonder which data set was used for training by the authors of MoleculeNet and other groups that used the `load_muv` function.

## To Reproduce

Steps to reproduce the behaviour:

1. Download the MUV data from https://moleculenet.org/datasets-1
```
> data = pd.read_csv(./muv.csv)
> data.isna().sum().sum()
1332593
```

2. Load data via deepchem.
```
> import deepchem as dc
> task, data, _ = dc.molnet.load_muv()
> train_dataset, val_dataset, test_dataset = data
> train_prop = train_dataset.y
> print(np.count_nonzero(np.isnan(train_prop)))
0
```

## Expected behavior

`load_muv()` should result in the same data as the `muv.csv` file.

## Environment
* DeepChem version: 2.7.1

**Comment by rbharath** [comment]
It is the same data; the weights matrix is used to mask out the missing data to avoid having NaNs in the numpy array.

**Comment by reyan-singh** [comment]
I think this may due to the way load_csv_files() in data_utils.py is taking care of nan and converting nan to string (see highlighted command below in screenshot)

<img width="723" alt="Screenshot 2024-03-30 at 7 28 50 PM" src="https://github.com/deepchem/deepchem/assets/121709625/1386e794-3114-48e1-90c3-fa5e52dc2285">

**Comment by egracheva** [comment]
Thank you for the replies! The "weight" in the name of the matrix confused me.
